### PR TITLE
TRACING-3360 | Distributed tracing 2.9 release notes

### PR DESCRIPTION
**Pull request by pavolloffay**
The preview is available at https://github.com/openshift/openshift-docs/pull/63263

Useful links:
* https://issues.redhat.com/secure/ReleaseNote.jspa?version=12407470&styleName=Html&projectId=12320921&Create=Create&atl_token=AQZJ-FV3A-N91S-UDEU_0107f4bcc06243839df494816b881b5f52a471f2_lin 
* https://issues.redhat.com/browse/TRACING-3337

Upstream changelog:
* OTEL: https://github.com/open-telemetry/opentelemetry-collector/releases 
* OTEL: https://github.com/open-telemetry/opentelemetry-collector-contrib/releases
* OTEL: https://github.com/open-telemetry/opentelemetry-operator
* Jaeger: https://github.com/jaegertracing/jaeger/releases
* Jaeger: https://github.com/jaegertracing/jaeger-operator/blob/main/CHANGELOG.md
* Tempo: https://github.com/grafana/tempo/releases
* Tempo: https://github.com/grafana/tempo-operator/releases

### CODE REVIEW

File: modules/distr-tracing-rn-new-features.adoc [hunk 1]
```diff
@@ -17,25 +17,13 @@ This release adds improvements related to the following components and concepts.
...
-
-
-
-
-
-
-
+* Support OTLP metrics ingestion in {OTELName}. The metrics can be forwarded and stored in the `+user-workload-monitoring+` deployment.
+* Bring {TempoName} operator to the level 4. This feature enables upgrade and monitoring of `+TempoStack+` instances and operator.
+* Bring {OTELName} operator to the level 4. This feature enables upgrade and monitoring of `+OpenTelemetryCollector+` instances and operator.
```
Comment (pavolloffay) [inline modules/distr-tracing-rn-new-features.adoc]: Add known issues about service monitors and level not being updated in the CSV.

File: modules/distr-tracing-rn-new-features.adoc [hunk 2]
```diff
@@ -17,25 +17,13 @@ This release adds improvements related to the following components and concepts.
...
-
-
-
-
-
-
-
-
+* Support OTLP metrics ingestion in {OTELName}. The metrics can be forwarded and stored in the `+user-workload-monitoring+` deployment.
+* Bring {TempoName} operator to the level 4. This feature enables upgrade and monitoring of `+TempoStack+` instances and operator.
```
Comment (pavolloffay) [inline modules/distr-tracing-rn-new-features.adoc]: add alerting

File: modules/distr-tracing-rn-new-features.adoc [hunk 3]
```diff
@@ -17,25 +17,13 @@ This release adds improvements related to the following components and concepts.
 
 This release of {DTProductName} addresses Common Vulnerabilities and Exposures (CVEs) and bug fixes.
 
+Notable enhancements:
```
Comment (pavolloffay) [inline modules/distr-tracing-rn-new-features.adoc]: Split to Tempo, OTEL and Jaeger. 

List only downstream issues.

Comment (pavolloffay) [reply modules/distr-tracing-rn-new-features.adoc]: create a section for known issues

**Review by frzifus — APPROVED** [review]

File: modules/distr-tracing-rn-new-features.adoc [hunk 4]
```diff
@@ -17,25 +17,53 @@ This release adds improvements related to the following components and concepts.
...
+* link:https://issues.redhat.com/browse/TRACING-3173[3173]: Jaeger operator pod restarting with OOMKilled with the default memory value.
 
+=== {TempoName}
 
+==== Enhancements
 
+* Support the operator level 4. This feature enables upgrade, monitoring and alerting of `+TempoStack+` instances and operator.
+* Add Ingress and OpenShift Route configuration for the Gateway.
+* Support managed and unmanaged state in the `+TempoStack+` custom resource.
+* Expose additional protocols (Jaeger Thrift binary, Jaeger Thrift compact, Jaeger gRPC and Zipkin) in the distributor service. When gateway is enabled only OTLP gRPC is enabled.
```
Comment (iblancasa) [inline modules/distr-tracing-rn-new-features.adoc]: Should we mention that the GRPC Query was also exposed?

**Review by iblancasa — APPROVED** [review]

File: modules/distr-tracing-rn-new-features.adoc [hunk 5]
```diff
@@ -17,25 +17,53 @@ This release adds improvements related to the following components and concepts.
...
+* Support multitenancy without the Gateway (authentication and authorization).
```
Comment (andreasgerstmayr) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Support multitenancy without the Gateway (without authentication and authorization).
```

Not sure if this additional word would make it more clear or more confusing

**Review by andreasgerstmayr — APPROVED** [review]

**Comment by pavolloffay** [comment]
@max-cx similar to logging we should add a link to errata advisory to the release notes https://docs.openshift.com/container-platform/4.13/logging/cluster-logging-release-notes.html#cluster-logging-release-notes-5-7-4_cluster-logging-release-notes-v5x. The advisory includes all fixed CVEs that I didn't put into changelog.

File: modules/distr-tracing-rn-new-features.adoc (hunk 4 above)
Comment (pavolloffay) [reply modules/distr-tracing-rn-new-features.adoc]: if it was I will include it

Comment (pavolloffay) [reply modules/distr-tracing-rn-new-features.adoc]: I have added it as a separate item https://github.com/grafana/tempo-operator/pull/517

**Comment by pavolloffay** [comment]
On the meeting with @max-cx we have discussed to have 2 levels of release notes:
1. level - include table with released components (like we do right now) and document there new features that span multiple components 
2. level - release notes per operator

File: modules/distr-tracing-rn-new-features.adoc [hunk 6]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
 
 This release of {DTProductName} addresses Common Vulnerabilities and Exposures (CVEs) and bug fixes.
 
+=== {JaegerName}
 
+==== Bug fixes
 
+* link:https://issues.redhat.com/browse/TRACING-3322[TRACING-3322]: Expose Jaeger Query gRPC port (16685) on the Jaeger Query service.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Expose Jaeger Query gRPC port (16685) on the Jaeger Query service.
// link:https://issues.redhat.com/browse/TRACING-3322[TRACING-3322]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 7]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
+* link:https://issues.redhat.com/browse/TRACING-3312[TRACING-3312]: When deploying Service Mesh on SNO in a disconnected environment , the Jaeger Pod frequently goes into Pending state.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* When deploying Service Mesh on SNO in a disconnected environment , the Jaeger Pod frequently goes into Pending state.
// link:https://issues.redhat.com/browse/TRACING-3312[TRACING-3312]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 8]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
+* link:https://issues.redhat.com/browse/TRACING-2968[TRACING-29638]: Wrong port is exposed for jaeger-production-query resulting in connection refused. Fixed by exposing Jaeger Query gRPC port (16685) on the Jaeger Query deployment.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Wrong port is exposed for jaeger-production-query resulting in connection refused. Fixed by exposing Jaeger Query gRPC port (16685) on the Jaeger Query deployment.
// link:https://issues.redhat.com/browse/TRACING-2968[TRACING-29638]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 9]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
+* link:https://issues.redhat.com/browse/TRACING-3173[TRACING-3173]: Jaeger operator pod restarting with OOMKilled with the default memory value. Fixed by removing resource limits.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Jaeger operator pod restarting with OOMKilled with the default memory value. Fixed by removing resource limits.
// link:https://issues.redhat.com/browse/TRACING-3173[TRACING-3173]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 10]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
+* Support the operator level 4 (Deep Insights). This feature enables upgrade, monitoring and alerting of `+TempoStack+` instances and operator.
+* Add Ingress and OpenShift Route configuration for the Gateway.
+* Support managed and unmanaged state in the `+TempoStack+` custom resource.
+* Expose additional ingestion protocols (Jaeger Thrift binary, Jaeger Thrift compact, Jaeger gRPC and Zipkin) in the Distributor service. When the Gateway is enabled only OTLP gRPC is enabled.
+* Expose Jaeger Query gRPC endpoint on the Query Frontend service.
+* Support multitenancy without the Gateway (without authentication and authorization).
 
+==== Bug fixes
 
+* link:https://issues.redhat.com/browse/TRACING-3145[TRACING-3145]: Support disconnected environments.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Support disconnected environments.
// link:https://issues.redhat.com/browse/TRACING-3145[TRACING-3145]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 11]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
+* link:https://issues.redhat.com/browse/TRACING-3091[TRACING-3091]: Enable mTLS communication between Tempo components.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Enable mTLS communication between Tempo components.
// link:https://issues.redhat.com/browse/TRACING-3091[TRACING-3091]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 12]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
+* link:https://issues.redhat.com/browse/TRACING-3204[TRACING-3204]: Remove resource limits for Tempo operator.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Remove resource limits for Tempo operator.
// link:https://issues.redhat.com/browse/TRACING-3204[TRACING-3204]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 13]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
 
+==== Known issues
 
+* link:https://issues.redhat.com/browse/TRACING-3462[TRACING-3462]: Custom TLS CA option is not implemented.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Custom TLS CA option is not implemented.
// link:https://issues.redhat.com/browse/TRACING-3462[TRACING-3462]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 14]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
+* link:https://issues.redhat.com/browse/TRACING-3139[TRACING-3139]: When you use the Jaeger user interface (UI) with Tempo Operator, the Jaeger UI lists only services that have sent traces within the last 15 minutes. For services that have not sent traces within the last 15 minutes, those traces are still stored even though they are not visible in the Jaeger UI.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* When you use the Jaeger user interface (UI) with Tempo Operator, the Jaeger UI lists only services that have sent traces within the last 15 minutes. For services that have not sent traces within the last 15 minutes, those traces are still stored even though they are not visible in the Jaeger UI.
// link:https://issues.redhat.com/browse/TRACING-3139[TRACING-3139]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 15]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
 
+* Support OTLP metrics ingestion. The metrics can be forwarded and stored in the `+user-workload-monitoring+` via the Prometheus exporter.
+* Support the operator level 4 (Deep Insights). This feature enables upgrade and monitoring of `+OpenTelemetryCollector+` instances and operator.
+* Report traces and metrics from remote clusters using OTLP/HTTP(s).
+* Collect OpenShift specific resource attributes via `+resourcedetection+` processor.
+* Support managed and unmanaged state in the `+OpenTelemetryCollector+` custom resouce.
 
+==== Bug fixes
 
+* link:https://issues.redhat.com/browse/TRACING-3190[TRACING-3190]: OpenTelemetry operator crashlooping after receiving opentelemetry-operator.v0.74.0-5. Fixed by properly configuring autoscaler.
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* OpenTelemetry operator crashlooping after receiving opentelemetry-operator.v0.74.0-5. Fixed by properly configuring autoscaler.
// link:https://issues.redhat.com/browse/TRACING-3190[TRACING-3190]
```

File: modules/distr-tracing-rn-new-features.adoc [hunk 16]
```diff
@@ -17,25 +17,58 @@ This release adds improvements related to the following components and concepts.
...
 
+==== Known issues
 
-
+* link:https://issues.redhat.com/browse/TRACING-3431[TRACING-3431]: Operator level needs to be set to 4 (Deep Insights).
```
Comment (max-cx) [inline modules/distr-tracing-rn-new-features.adoc]:
```suggestion
* Operator level needs to be set to 4 (Deep Insights).
// link:https://issues.redhat.com/browse/TRACING-3431[TRACING-3431]
```

File: modules/distr-tracing-rn-known-issues.adoc [hunk 17]
```diff
@@ -44,8 +44,6 @@ These are the known issues for {DTProductName}:
 
 
 
-* link:https://issues.redhat.com/browse/OBSDA-220[OBSDA-220] In some cases, if you try to pull an image using {OTELShortName}, the image pull fails and a `Failed to pull image` error message appears. 
-There is no workaround for this issue.
 
 * link:https://issues.redhat.com/browse/TRACING-2057[TRACING-2057] The Kafka API has been updated to `v1beta2` to support the Strimzi Kafka Operator 0.23.0. However, this API version is not supported by AMQ Streams 1.6.3. If you have the following environment, your Jaeger services will not be upgraded, and you cannot create new Jaeger services or modify existing Jaeger services:
```
Comment (max-cx) [inline modules/distr-tracing-rn-known-issues.adoc:48]:
```suggestion
* The Kafka API has been updated to `v1beta2` to support the Strimzi Kafka Operator 0.23.0. However, this API version is not supported by AMQ Streams 1.6.3. If you have the following environment, your Jaeger services will not be upgraded, and you cannot create new Jaeger services or modify existing Jaeger services:
// link:https://issues.redhat.com/browse/TRACING-2057[TRACING-2057]
```

Comment (pavolloffay) [reply modules/distr-tracing-rn-known-issues.adoc:48]: I will leave this out as it was not subject of this PR. There are probably other links in the changelog that would need to be removed.

File: modules/distr-tracing-rn-new-features.adoc [hunk 18]
```diff
@@ -17,25 +17,125 @@ This release adds improvements related to the following components and concepts.
...
+// link:https://issues.redhat.com/browse/TRACING-3462[TRACING-3462]
+
+* When you use the Jaeger user interface (UI) with Tempo Operator, the Jaeger UI lists only services that have sent traces within the last 15 minutes.
+For services that have not sent traces within the last 15 minutes, those traces are still stored even though they are not visible in the Jaeger UI.
+
+// link:https://issues.redhat.com/browse/TRACING-3139[TRACING-3139]
+
+* Tempo query frontend service should not use internal mTLS when gateway is not deployed. This issue does not affect Jaeger Query API. The workaround is to disable mTLS.
+
+.Disable mTLS
```
Comment (pavolloffay) [inline modules/distr-tracing-rn-new-features.adoc:84]: @rubenvp8510 could you please review the procedure to disable the mTLS? It ends on line 110.

Comment (max-cx) [reply modules/distr-tracing-rn-new-features.adoc:84]: Responded by Slack.

Comment (rubenvp8510) [reply modules/distr-tracing-rn-new-features.adoc:84]: This procedure looks good!

Comment (IshwarKanse) [reply modules/distr-tracing-rn-new-features.adoc:84]: Verified the steps to disable the mTLS.

**Pull request merged**